static/app/views/settings/organizationRepositories/index.tsx

```diff
@@ -1,61 +1,74 @@
 import {t} from 'sentry/locale';
-import {Organization, Repository} from 'sentry/types';
+import {Repository} from 'sentry/types';
+import {useApiQuery} from 'sentry/utils/queryClient';
 import routeTitleGen from 'sentry/utils/routeTitle';
-import withOrganization from 'sentry/utils/withOrganization';
-import DeprecatedAsyncView from 'sentry/views/deprecatedAsyncView';
+import {useLocation} from 'sentry/utils/useLocation';
+import useOrganization from 'sentry/utils/useOrganization';
 
 import OrganizationRepositories from './organizationRepositories';
 
-type Props = RouteComponentProps<{}, {}> & {
-  organization: Organization;
-} & DeprecatedAsyncView['props'];
+function OrganizationRepositoriesContainer() {
+  const organization = useOrganization();
+  const location = useLocation();
 
-type State = DeprecatedAsyncView['state'] & {
-  itemList: Repository[] | null;
-};
+  const {
+    data: itemData,
+    isLoading,
+    isError,
+    getResponseHeader,
+  } = useApiQuery<Repository[]>(
+    [`/organizations/${organization.slug}/repos/`, {query: location.query}],
+    {staleTime: 0}
+  );
+  const itemListPageLinks = getResponseHeader?.('Link');
+  const [itemList, setItemList] = useState<Repository[] | null>(null);
 
-class OrganizationRepositoriesContainer extends DeprecatedAsyncView<Props, State> {
-  getEndpoints(): ReturnType<DeprecatedAsyncView['getEndpoints']> {
-    const {organization} = this.props;
-    return [['itemList', `/organizations/${organization.slug}/repos/`]];
+  useEffect(() => {
+    setItemList(null);
+  }, [location.query]);
+
+  if (isLoading) {
+    return <LoadingIndicator />;
   }
 
-  // Callback used by child component to signal state change
-  onRepositoryChange = (data: Pick<Repository, 'id' | 'status'>) => {
-    const itemList = this.state.itemList;
-    itemList?.forEach(item => {
-      if (item.id === data.id) {
-        item.status = data.status;
-      }
-    });
-    this.setState({itemList});
-  };
-
-  getTitle() {
-    const {organization} = this.props;
-    return routeTitleGen(t('Repositories'), organization.slug, false);
+  if (isError) {
+    return <LoadingError />;
+  }
+
+  if (itemData && !itemList) {
+    setItemList(itemData);
   }
 
-  renderBody() {
-    const {itemList, itemListPageLinks} = this.state;
-
-    return (
-      <Fragment>
-        <OrganizationRepositories
-          {...this.props}
-          itemList={itemList!}
-          onRepositoryChange={this.onRepositoryChange}
-        />
-        {itemListPageLinks && (
-          <Pagination pageLinks={itemListPageLinks} {...this.props} />
-        )}
-      </Fragment>
-    );
+  // Callback used by child component to signal state change
+  function onRepositoryChange(data: Pick<Repository, 'id' | 'status'>) {
+    if (itemList) {
+      setItemList(
+        itemList.map(item =>
+          item.id === data.id ? {...item, status: data.status} : item
+        )
+      );
+    }
   }
```

Comment:
Instead of putting state here I think we want to use the react-query `useMutation` stuff. cc @malwilley

Reply:
That's right - before, AsyncComponent maintained state so you needed to call `setState` to update when making modifications. The right way to deal with things now is to update the query cache, since that is the source of truth when using useApiQuery.

There's some info in the docs about this: https://develop.sentry.dev/frontend/network-requests/#updating-your-query-data

The most simple way to do this would be to just update the cache here in `onRepositoryChange`. The _best_ way would be to update the cache the the sub-component where the update requests are being made, but that requires more changes so I wouldn't expect you to go that far for this conversion.